packages/core-mobile/app/screens/bridge/Bridge.tsx

```diff
@@ -86,261 +65,276 @@ const Bridge: FC = () => {
   const dispatch = useDispatch()
-  const criticalConfig = useSelector(selectBridgeCriticalConfig)
-  const [selectedAsset, setSelectedAsset] = useState<AssetBalance>()
+  const { activeNetwork } = useNetworks()
+
   const selectedCurrency = useSelector(selectSelectedCurrency)
   const {
-    sourceBalance,
-    amount,
-    setAmount,
+    assetBalance,
+    inputAmount,
+    setInputAmount,
     assetsWithBalances,
     networkFee,
     loading,
     price,
     maximum,
     minimum,
-    receiveAmount,
-    wrapStatus,
     transfer,
     bridgeFee,
-    provider,
-    denomination,
-    amountBN
-  } = useBridge(selectedAsset)
+    bridgeType,
+    amount,
+    sourceNetworks,
+    targetNetworks,
+    sourceNetwork,
+    setSourceNetwork,
+    targetNetwork,
+    setTargetNetwork,
+    bridgeAssets,
+    selectedBridgeAsset,
+    setSelectedBridgeAsset,
+    error
+  } = useBridge()
 
-  const {
-    setCurrentAsset: setCurrentAssetSymbol,
-    currentBlockchain,
-    targetBlockchain
-  } = useBridgeSDK()
-  const { activeNetwork, networks } = useNetworks()
   const activeAccount = useSelector(selectActiveAccount)
-  const { getTokenSymbolOnNetwork } = useGetTokenSymbolOnNetwork()
   const [bridgeError, setBridgeError] = useState('')
   const [isPending, setIsPending] = useState(false)
-  const tokenInfoData = useTokenInfoContext()
   const nativeTokenBalance = useSelector((state: RootState) =>
     selectAvailableNativeTokenBalanceForNetworkAndAccount(
       state,
       activeNetwork.chainId,
       activeAccount?.index
     )
   )
-  const selectedAssetSymbol = useMemo(
-    () =>
-      isUnifiedBridgeAsset(selectedAsset?.asset)
-        ? selectedAsset?.asset.symbol
-        : getTokenSymbolOnNetwork(
-            selectedAsset?.asset.symbol ?? '',
-            currentBlockchain
-          ),
-    [currentBlockchain, getTokenSymbolOnNetwork, selectedAsset?.asset]
-  )
+  const selectedAssetSymbol = selectedBridgeAsset?.symbol
 
-  const { bridgeBtcBlocked, bridgeEthBlocked } = usePosthogContext()
   const { currencyFormatter } = useApplicationContext().appHook
-  const isAmountTooLow =
-    amount && !amount.eq(BIG_ZERO) && minimum && amount.lt(minimum)
-
-  const isAmountTooLarge =
-    amount && !amount.eq(BIG_ZERO) && maximum && amount.gt(maximum)
-
+  const isAmountTooLow = amount !== 0n && minimum && amount < minimum
+  const isAmountTooLarge = amount !== 0n && maximum && amount > maximum
   const isNativeBalanceNotEnoughForNetworkFee = Boolean(
-    amount &&
-      !amount.eq(BIG_ZERO) &&
+    amount !== 0n &&
       networkFee &&
-      bigintToBig(nativeTokenBalance, activeNetwork.networkToken.decimals).lt(
-        networkFee
-      )
+      nativeTokenBalance <
+        networkFee +
+          (assetBalance?.asset.type === TokenType.NATIVE ? amount : 0n)
   )
 
-  const hasValidAmount = !isAmountTooLow && amount.gt(BIG_ZERO)
+  const denomination = useMemo(() => {
+    if (!assetBalance) {
+      return 0
```

Comment:
does 0 really work? if not, maybe better to throw an error?

Reply:
also I feel like some of the useMemo is not necessary since there is no heavy computation happening.